lib/Futures/CMakeLists.txt

```diff
@@ -2,8 +2,12 @@ add_library(arango_futures STATIC
   src/Future.cpp)
 
 target_include_directories(arango_futures
-  PUBLIC
-  ${PROJECT_SOURCE_DIR}/3rdParty/iresearch/external/function2
-  include)
+        PUBLIC
+        ${PROJECT_SOURCE_DIR}/3rdParty/iresearch/external/function2
+        include)
+
+target_include_directories(arango_futures
+        PUBLIC
+        ${PROJECT_SOURCE_DIR}/lib/Inspection/include)
```

Comment:
Include was necessary, because `ResultT` headers use some `Inspector` types to make it serializable.

Reply:
Can you not just use 
```
target_link_libraries(arango_futures, INTERFACE arango_inspection)
``` 
instead of that hack?